### PR TITLE
Drop support for Python 3.5

### PR DESCRIPTION
**Pull request by srittau**
Python 3.5 EOL was on 2020-09-30.

### CODE REVIEW

File: scripts/migrate_script.py [hunk 1]
```diff
@@ -113,7 +113,7 @@ def collect_stdlib_packages() -> Tuple[List[StdLibPackage], List[StdLibPackage]]
     add_stdlib_packages_from("stdlib/2and3", stdlib, "2.7")
     # Use oldest currently supported version for Python 3 packages/modules.
     add_stdlib_packages_from("stdlib/3", stdlib, DEFAULT_PY3_VERSION)
-    for version in ("3.6", "3.7", "3.8", "3.9"):
+    for version in ("3.7", "3.8", "3.9"):
```
Comment (CraftSpider) [inline scripts/migrate_script.py:116]: This looks to be removing 3.6, not 3.5. Is there a reason for not support stdlib3.6 when removing 3.5?

Comment (srittau) [reply scripts/migrate_script.py:116]: This iterates over the directories in strong/third-party. But the 3.6 directory has now been merged into the 3 directory.

**Review by hauntsaninja — COMMENTED** [review]
Thanks for doing this! Looks good as far as I can tell.

I don't have a good sense of when the right time to start burying 3.5's corpse is, so trusting you on that. It looks like we waited four months past EOL for 3.4.
Note @JukkaL expressed some reluctance recently: https://github.com/python/mypy/issues/9536#issuecomment-706109850 For mypy, 3.5 seems to account for only a small share of downloads (although we don't know what version of Python people are targetting).
In terms of precedent setting, time will tell, but I think we might want to be a little slower when 3.6's time comes.

**Comment by JukkaL** [comment]
The decision to drop 3.5 support in typeshed can be orthogonal to the decision to drop 3.5 support in mypy, I think. Type checking Python 3.5 code will be possible (though somewhat limited) even without explicit support in typeshed. Mypy just won't be able to flag uses of Python 3.6 library features when targeting 3.5. Almost all changes in 3.6 seem to add new functionality instead of removing existing functionality or modifying things in incompatible ways. 

Starting from the next mypy release, I'm fine with 3.5 support being incomplete. The main thing is being able to run mypy using Python 3.5 when that's what ships with the operating system.

File: scripts/migrate_script.py (hunk 1 above)
Comment (CraftSpider) [reply scripts/migrate_script.py:116]: Ahh, right

**Comment by JelleZijlstra** [comment]
I'm OK with doing this, but there's a ton of merge conflicts.

**Comment by srittau** [comment]
@JelleZijlstra I rebased the PR on current master.

**Review by JelleZijlstra — APPROVED** [review]
Looks good, but do you know what's going on with mypy-primer here?

**Pull request merged**